textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -33,12 +35,15 @@
     private final KeywordPredicate predicate;
     private IOperator inputOperator;
     private String query;
-
+    
     public KeywordMatcher(IPredicate predicate) {
         this.predicate = (KeywordPredicate)predicate;
-        DataReaderPredicate dataReaderPredicate = this.predicate.getDataReaderPredicate();
-        dataReaderPredicate.setIsSpanInformationAdded(true);
-        this.inputOperator = new IndexBasedSourceOperator(dataReaderPredicate);
+        this.query = this.predicate.getQuery();
+    }
+    
+    public KeywordMatcher(IPredicate predicate, IOperator inputOperator) {
```

Comment:
Let's always require the caller to call `setInputOperator()` explicitly to build the connection.